overlord/ifacestate/helpers_test.go

```diff
@@ -417,6 +418,126 @@ apps:
 	c.Check(osutil.FileExists(dirs.SnapSystemKeyFile), Equals, false)
 }
 
+func mockSnaps(c *C, st *state.State) {
+	// Put fake snaps in the state
+	for _, name := range []string{"foo", "bar"} {
+		yamlText := `
+name: %NAME%
+version: 1
+apps:
+  test:
+    command: bin/test
+`
+		si := &snap.SideInfo{Revision: snap.R(1), RealName: name}
+		snapInfo := snaptest.MockSnap(c, strings.Replace(yamlText, "%NAME%", name, -1), si)
+		st.Lock()
+		snapst := &snapstate.SnapState{
+			SnapType: string(snap.TypeApp),
+			Sequence: []*snap.SideInfo{si},
+			Active:   true,
+			Current:  snap.R(1),
+		}
+		snapstate.Set(st, snapInfo.InstanceName(), snapst)
+		st.Unlock()
+	}
+}
+
+func (s *helpersSuite) TestProfileRegenerationSetupMany(c *C) {
+	dirs.SetRootDir(c.MkDir())
+	defer dirs.SetRootDir("")
+
+	var setupManyCalls int
+	var writeKey bool
+
+	// Create a fake security backend
+	backend := &ifacetest.TestSecurityBackendSetupMany{
+		TestSecurityBackend: ifacetest.TestSecurityBackend{BackendName: "fake"},
+		SetupManyCallback: func(snaps []*snap.Info, confinement func(snapName string) interfaces.ConfinementOptions, repo *interfaces.Repository, tm timings.Measurer) []error {
+			c.Check(snaps, HasLen, 2)
+			setupManyCalls++
+			return nil
+		},
+	}
+	restore := ifacestate.MockSecurityBackends([]interfaces.SecurityBackend{backend})
+	defer restore()
+
+	// Create a mock overlord, mainly to have state.
+	ovld := overlord.Mock()
+	st := ovld.State()
+
+	mockSnaps(c, st)
+
+	// Pretend that security profiles are out of date and mock the
+	// function that writes the new system key with one always panics.
```

Comment:
where does the function panic here?

Reply:
Thanks, tweaked all the comments you pointed out.